router/router.go

```diff
@@ -90,6 +90,10 @@ func (self *Router) MetricsRegistry() metrics.UsageRegistry {
 }
 
 func (self *Router) Channel() channel2.Channel {
+	// if we're just starting up, we may be nil. wait till initialized
+	for self.ctrl == nil {
```

Comment:
should we have some MAX time after which we stop the process?

Reply:
If the control channel isn't established within a certain amount of time, the router will exit

Reply:
some other code is handling that?

Reply:
will do. the initial control channel connect has a timeout

Reply:
Added comment